openvdb/math/Coord.h

```diff
@@ -376,7 +376,17 @@ class CoordBBox
     bool operator!=(const CoordBBox& rhs) const { return !(*this == rhs); }
 
     /// @brief Return @c true if this bounding box is empty (i.e., encloses no coordinates).
-    bool empty() const { return (mMin[0] > mMax[0] || mMin[1] > mMax[1] || mMin[2] > mMax[2]); }
+    bool empty() const
+    {
+#if defined(__GNUC__)
+  #pragma GCC diagnostic push
+  #pragma GCC diagnostic ignored "-Wstrict-overflow"
+#endif
+        return (mMin[0] > mMax[0] || mMin[1] > mMax[1] || mMin[2] > mMax[2]);
+#if defined(__GNUC__)
+  #pragma GCC diagnostic pop
+#endif
+    }
```

Comment:
Had trouble trying to resolve this :|

Reply:
Yeah, I had a go at this too, I couldn't even figure out where this was coming from, the stack trace is pretty obscure! This seems like a reasonable compromise.

Reply:
ICC, for one, defines `__GNUC__`, but it uses different diagnostic flags, so at a minimum you'll need to `#ifndef __INTEL_COMPILER` (or something like that).

What _was_ the stack trace, though?

Reply:
See below for warnings from the core lib

<details><summary>Stack trace</summary>
<p>

```
In file included from ../openvdb/Types.h:45:0,
                 from ../openvdb/openvdb.h:35,
                 from python/pyGrid.h:62,
                 from python/pyIntGrid.cc:34:
../openvdb/math/Coord.h: In member function ‘void openvdb::v6_0abi5::tree::InternalNode<_ChildNodeType, Log2Dim>::clip(const openvdb::v6_0abi5::math::CoordBBox&, const ValueType&) [with _ChildNodeType = openvdb::v6_0abi5::tree::InternalNode<openvdb::v6_0abi5::tree::LeafNode<bool, 3u>, 4u>; unsigned int Log2Dim = 5u]’:
../openvdb/math/Coord.h:381:35: warning: assuming signed overflow does not occur when assuming that (X + c) < X is always false [-Wstrict-overflow]
         return (mMin[0] > mMax[0] || mMin[1] > mMax[1] || mMin[2] > mMax[2]);
                 ~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~
../openvdb/math/Coord.h:381:56: warning: assuming signed overflow does not occur when assuming that (X + c) < X is always false [-Wstrict-overflow]
         return (mMin[0] > mMax[0] || mMin[1] > mMax[1] || mMin[2] > mMax[2]);
                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~
../openvdb/math/Coord.h:381:56: warning: assuming signed overflow does not occur when assuming that (X + c) < X is always false [-Wstrict-overflow]
         return (mMin[0] > mMax[0] || mMin[1] > mMax[1] || mMin[2] > mMax[2]);
                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~
../openvdb/math/Coord.h: In member function ‘void openvdb::v6_0abi5::tree::Tree<_RootNodeType>::readBuffers(std::istream&, const openvdb::v6_0abi5::math::CoordBBox&, bool) [with _RootNodeType = openvdb::v6_0abi5::tree::RootNode<openvdb::v6_0abi5::tree::InternalNode<openvdb::v6_0abi5::tree::InternalNode<openvdb::v6_0abi5::tree::LeafNode<bool, 3u>, 4u>, 5u> >]’:
../openvdb/math/Coord.h:381:35: warning: assuming signed overflow does not occur when assuming that (X + c) < X is always false [-Wstrict-overflow]
         return (mMin[0] > mMax[0] || mMin[1] > mMax[1] || mMin[2] > mMax[2]);
                 ~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~
../openvdb/math/Coord.h:381:56: warning: assuming signed overflow does not occur when assuming that (X + c) < X is always false [-Wstrict-overflow]
         return (mMin[0] > mMax[0] || mMin[1] > mMax[1] || mMin[2] > mMax[2]);
                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~
../openvdb/math/Coord.h:381:56: warning: assuming signed overflow does not occur when assuming that (X + c) < X is always false [-Wstrict-overflow]
         return (mMin[0] > mMax[0] || mMin[1] > mMax[1] || mMin[2] > mMax[2]);
                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~
../openvdb/math/Coord.h:381:35: warning: assuming signed overflow does not occur when assuming that (X + c) < X is always false [-Wstrict-overflow]
         return (mMin[0] > mMax[0] || mMin[1] > mMax[1] || mMin[2] > mMax[2]);
                 ~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~
../openvdb/math/Coord.h:381:56: warning: assuming signed overflow does not occur when assuming that (X + c) < X is always false [-Wstrict-overflow]
         return (mMin[0] > mMax[0] || mMin[1] > mMax[1] || mMin[2] > mMax[2]);
                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~
../openvdb/math/Coord.h:381:56: warning: assuming signed overflow does not occur when assuming that (X + c) < X is always false [-Wstrict-overflow]
         return (mMin[0] > mMax[0] || mMin[1] > mMax[1] || mMin[2] > mMax[2]);
                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~
../openvdb/math/Coord.h: In member function ‘void openvdb::v6_0abi5::Grid< <template-parameter-1-1> >::readBuffers(std::istream&, const openvdb::v6_0abi5::math::CoordBBox&) [with _TreeType = openvdb::v6_0abi5::tree::Tree<openvdb::v6_0abi5::tree::RootNode<openvdb::v6_0abi5::tree::InternalNode<openvdb::v6_0abi5::tree::InternalNode<openvdb::v6_0abi5::tree::LeafNode<bool, 3u>, 4u>, 5u> > >]’:
../openvdb/math/Coord.h:381:35: warning: assuming signed overflow does not occur when assuming that (X + c) < X is always false [-Wstrict-overflow]
         return (mMin[0] > mMax[0] || mMin[1] > mMax[1] || mMin[2] > mMax[2]);
                 ~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~
../openvdb/math/Coord.h:381:56: warning: assuming signed overflow does not occur when assuming that (X + c) < X is always false [-Wstrict-overflow]
         return (mMin[0] > mMax[0] || mMin[1] > mMax[1] || mMin[2] > mMax[2]);
                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~
../openvdb/math/Coord.h:381:56: warning: assuming signed overflow does not occur when assuming that (X + c) < X is always false [-Wstrict-overflow]
         return (mMin[0] > mMax[0] || mMin[1] > mMax[1] || mMin[2] > mMax[2]);
                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~
../openvdb/math/Coord.h:381:35: warning: assuming signed overflow does not occur when assuming that (X + c) < X is always false [-Wstrict-overflow]
         return (mMin[0] > mMax[0] || mMin[1] > mMax[1] || mMin[2] > mMax[2]);
                 ~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~
../openvdb/math/Coord.h:381:56: warning: assuming signed overflow does not occur when assuming that (X + c) < X is always false [-Wstrict-overflow]
         return (mMin[0] > mMax[0] || mMin[1] > mMax[1] || mMin[2] > mMax[2]);
                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~
../openvdb/math/Coord.h:381:56: warning: assuming signed overflow does not occur when assuming that (X + c) < X is always false [-Wstrict-overflow]
         return (mMin[0] > mMax[0] || mMin[1] > mMax[1] || mMin[2] > mMax[2]);
                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~

In file included from ../openvdb/Types.h:45:0,
                 from ../openvdb/openvdb.h:35,
                 from viewer/RenderModules.h:34,
                 from viewer/RenderModules.cc:31:
../openvdb/math/Coord.h: In member function ‘void openvdb::v6_0abi5::tree::Tree<_RootNodeType>::readBuffers(std::istream&, const openvdb::v6_0abi5::math::CoordBBox&, bool) [with _RootNodeType = openvdb::v6_0abi5::tree::RootNode<openvdb::v6_0abi5::tree::InternalNode<openvdb::v6_0abi5::tree::InternalNode<openvdb::v6_0abi5::tree::LeafNode<bool, 3u>, 4u>, 5u> >]’:
../openvdb/math/Coord.h:381:35: warning: assuming signed overflow does not occur when assuming that (X + c) < X is always false [-Wstrict-overflow]
         return (mMin[0] > mMax[0] || mMin[1] > mMax[1] || mMin[2] > mMax[2]);
                 ~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~
../openvdb/math/Coord.h:381:56: warning: assuming signed overflow does not occur when assuming that (X + c) < X is always false [-Wstrict-overflow]
         return (mMin[0] > mMax[0] || mMin[1] > mMax[1] || mMin[2] > mMax[2]);
                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~
../openvdb/math/Coord.h:381:56: warning: assuming signed overflow does not occur when assuming that (X + c) < X is always false [-Wstrict-overflow]
         return (mMin[0] > mMax[0] || mMin[1] > mMax[1] || mMin[2] > mMax[2]);
                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~
../openvdb/math/Coord.h:381:35: warning: assuming signed overflow does not occur when assuming that (X + c) < X is always false [-Wstrict-overflow]
         return (mMin[0] > mMax[0] || mMin[1] > mMax[1] || mMin[2] > mMax[2]);
                 ~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~
../openvdb/math/Coord.h:381:56: warning: assuming signed overflow does not occur when assuming that (X + c) < X is always false [-Wstrict-overflow]
         return (mMin[0] > mMax[0] || mMin[1] > mMax[1] || mMin[2] > mMax[2]);
                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~
../openvdb/math/Coord.h:381:56: warning: assuming signed overflow does not occur when assuming that (X + c) < X is always false [-Wstrict-overflow]
         return (mMin[0] > mMax[0] || mMin[1] > mMax[1] || mMin[2] > mMax[2]);
                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~

```

</p>
</details>

Reply:
I've added a check for `__INTEL_COMPILER`